plugins/outputs/newrelic/newrelic.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+)
+
+// NewRelic nr structure
+type NewRelic struct {
+	harvestor    *telemetry.Harvester
+	dc           *cumulative.DeltaCalculator
+	InsightsKey  string            `toml:"insights_key"`
+	MetricPrefix string            `toml:"metric_prefix"`
+	Timeout      internal.Duration `toml:"timeout"`
+	savedErrors  map[int]interface{}
+	errorCount   int
+	Client       http.Client
+}
+
+// Description returns a one-sentence description on the Output
+func (nr *NewRelic) Description() string {
+	return "Send metrics to New Relic metrics endpoint"
+}
+
+// SampleConfig : return  default configuration of the Output
+func (nr *NewRelic) SampleConfig() string {
+	return `
+	## New Relic Insights API key (required)
+	insights_key = "insights api key"
+
+	# metric_prefix if defined, prefix's metrics name for easy identification (optional)
+	# metric_prefix = ""
+	
+	# harvest timeout, default is 15 seconds
+	# timeout = "15s"
+`
+}
+
+// Connect to the Output
+func (nr *NewRelic) Connect() error {
+	if nr.InsightsKey == "" {
+		return fmt.Errorf("InsightKey is a required for newrelic")
+	}
+	var err error
+	nr.harvestor, err = telemetry.NewHarvester(telemetry.ConfigAPIKey(nr.InsightsKey),
+		telemetry.ConfigHarvestPeriod(0),
+		func(cfg *telemetry.Config) {
+			cfg.Product = "NewRelic-Telegraf-Plugin"
+			cfg.ProductVersion = "1.0"
+			cfg.HarvestTimeout = nr.Timeout.Duration
+			cfg.Client = &nr.Client
+			cfg.ErrorLogger = func(e map[string]interface{}) {
+				var errorString string
+				for k, v := range e {
+					errorString += fmt.Sprintf("%s = %s ", k, v)
+				}
+				nr.errorCount++
+				nr.savedErrors[nr.errorCount] = errorString
+			}
+		})
+	if err != nil {
+		return fmt.Errorf("unable to connect to newrelic %v", err)
+	}
+
+	nr.dc = cumulative.NewDeltaCalculator()
+	return nil
+}
+
+// Close any connections to the Output
+func (nr *NewRelic) Close() error {
+
+	nr.errorCount = 0
+	nr.Client.CloseIdleConnections()
+	return nil
+}
+
+// Write takes in group of points to be written to the Output
+func (nr *NewRelic) Write(metrics []telegraf.Metric) error {
+
+	nr.errorCount = 0
+	nr.savedErrors = make(map[int]interface{})
+
+	for _, metric := range metrics {
+		// create tag map
+		tags := make(map[string]interface{})
+		for _, tag := range metric.TagList() {
+			tags[tag.Key] = tag.Value
+		}
+		for _, field := range metric.FieldList() {
+			var mvalue float64
+			var mname string
+			if nr.MetricPrefix != "" {
+				mname = nr.MetricPrefix + "." + metric.Name() + "." + field.Key
+			} else {
+				mname = metric.Name() + "." + field.Key
+			}
+			switch n := field.Value.(type) {
+			case int64:
+				mvalue = float64(n)
+			case uint64:
+				mvalue = float64(n)
+			case float64:
+				mvalue = float64(n)
+			case bool:
+				mvalue = float64(0)
+				if n {
+					mvalue = float64(1)
+				}
+			case string:
+				// Do not log everytime we encounter string
+				// we just skip
```

Comment:
why discard strings? New Relic seems to support them just fine. If it's about saving space, this might be better off as a configuration option.

Reply:
@ssoroka  newrelic-telemetry-sdk supports float64 as value for the metric, so we are converting each value to float64.  We do add tag values in the attributes for the metric. Since strings are not supported as metric values we can document it in the README and revisit once it is supported by new relic
https://godoc.org/github.com/newrelic/newrelic-telemetry-sdk-go/telemetry#Gauge.

Reply:
Why not just store them as part of the attributes?

Reply:
Attributes in New relic are similar to Tags in the Telegraf. If you think we must, We can add them as attributes, the just would not have functionality related to metric visualization and querying.

Reply:
No, just curious. 👍